cephfs/permissions_test.go

```diff
@@ -64,3 +65,52 @@ func TestChown(t *testing.T) {
 	assert.Equal(t, uint32(sx.Uid), bob)
 	assert.Equal(t, uint32(sx.Gid), bob)
 }
+
+func TestLchown(t *testing.T) {
+	dirname := "four"
+	fname := "TestFile.txt"
+	var bob uint32 = 1010
+	var root uint32
+
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+	f1, err := mount.Open(fname, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	assert.NoError(t, err)
+	assert.NotNil(t, f1)
+
+	err = mount.Symlink(fname, "Symlnk")
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, mount.Unlink("Symlnk"))
+	}()
+
+	err = mount.Lchown(fname, bob, bob)
+	sx, err := mount.Statx("Symlnk", StatxBasicStats, 0)
```

Comment:
These steps appear backwards to me. The Lchown function does not follow symlinks, but when applied to regular files (fname) it should behave the same as Chown. Please reverse the files that these functions are called on and fix up the assertions - stat both fname and "Symlnk" to see that the ownership changes only for Symlnk. You will probably need to use the AtSymlinkNofollow flag with the Statx calls. Once that's done, I think we'll have a good test for Lchown. Thanks!

Reply:
> These steps appear backwards to me. The Lchown function does not follow symlinks, but when applied to regular files (fname) it should behave the same as Chown. Please reverse the files 

Sure, thanks, I am a bit confused probably by the wordings here, so you mean instead of calling `Lchown` on te file, I should call it on the path where I created the symlink?.

> that these functions are called on and fix up the assertions - stat both fname and "Symlnk" to see that the ownership changes only for Symlnk. You will probably need to use the `AtSymlinkNofollow` flag 

I will have to dig about `AtSymlinkNofollow`, I can see it's been used in `directory_test.go` so I think I can take a reference from there and fix the PR.

> with the Statx calls. Once that's done, I think we'll have a good test for Lchown. Thanks!

Reply:
> > These steps appear backwards to me. The Lchown function does not follow symlinks, but when applied to regular files (fname) it should behave the same as Chown. Please reverse the files
> 
> Sure, thanks, I am a bit confused probably by the wordings here, so you mean instead of calling `Lchown` on te file, I should call it on the path where I created the symlink?.

No worries. Yes, use the path of the symlink. FWIW in your current code the (relative) path of the symlink is "Symlnk". Don't get it confused with the path of the symlink's parent directory.

> > that these functions are called on and fix up the assertions - stat both fname and "Symlnk" to see that the ownership changes only for Symlnk. You will probably need to use the `AtSymlinkNofollow` flag
> 
> I will have to dig about `AtSymlinkNofollow`, I can see it's been used in `directory_test.go` so I think I can take a reference from there and fix the PR.
> 

Sure - in short, it tells Statx not to follow symlinks to their destinations but only report the stats of the symlink itself. If it helps his behavior is analogous to the AT_SYMLINK_NOFOLLOW for [fstatat file systems api](https://man7.org/linux/man-pages/man2/lstat.2.html). The behavior of the flags for this call and libcephfs are meant to be similar.